### PR TITLE
WIP Chrome dev tools

### PR DESCRIPTION
**Pull request by joelwurtz**
Goal is to add pure functional test and js execution into asynit

### CODE REVIEW

File: src/Extension/Chrome/Browser.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                'method' => $method,
+                'params' => $params
+            ]);
+
+            $this->nextMessageId++;
+
+            $this->logger->info('Send message ' . $message);
+
+            yield $this->connection->send($message);
+        });
```
Comment (kelunik) [inline src/Extension/Chrome/Browser.php]: Any exception inside this `Amp\call` will go unnoticed. It's better to use `Amp\asyncCall` to have them thrown into the loop or better just handle them.

File: composer.json [hunk 2]
```diff
@@ -29,7 +29,8 @@
     "require": {
         "php": "^7.0",
         "amphp/artax": "^3.0",
-        "amphp/websocket": "dev-master",
+        "amphp/websocket": "^0.2",
+        "amphp/process": "^0.2",
```
Comment (kelunik) [inline composer.json]: Why `0.2` instead of `0.3`?

File: example/chrome-example.php [hunk 3]
```diff
@@ -4,12 +4,37 @@
...
+                "about:blank"
+            ]
+        );
+
+        $process->start();
+        $found = false;
+        $url = "";
+
+        while (!$found && $process->isRunning()) {
+            $readed = yield $process->getStderr()->read();
```
Comment (kelunik) [inline example/chrome-example.php]: I guess this should be `$readed .=`, apart from being named `$read` :stuck_out_tongue_winking_eye:

File: example/chrome-example.php [hunk 4]
```diff
@@ -4,12 +4,37 @@
...
-// @TODO Change me
-$url = 'ws://127.0.0.1:9222/devtools/browser/b1c46ece-85c4-48b5-850d-687296d0566e';
 
-\Amp\Loop::run(function () use($url) {
+\Amp\Loop::run(function () {
     try {
-        $logger = new \Symfony\Component\Console\Logger\ConsoleLogger(new \Symfony\Component\Console\Output\ConsoleOutput(\Symfony\Component\Console\Output\OutputInterface::VERBOSITY_NORMAL));
+        $process = new \Amp\Process\Process(
+            [
+                "google-chrome-stable",
```
Comment (kelunik) [inline example/chrome-example.php]: I think this should be `google-chrome` instead, which should also exist if `google-chrome-stable` exists.

File: composer.json (hunk 2 above)
Comment (joelwurtz) [reply composer.json]: conflict with artax afaik

File: example/chrome-example.php (hunk 4 above)
Comment (joelwurtz) [reply example/chrome-example.php]: Depends on the env, will refactor this latter, right now i just want something that work on my differents env, so i can try / work on the api

**Comment by joelwurtz** [comment]
Useless

**Pull request closed without merging**